### PR TITLE
Add support of Jsoniter Scala

### PR DESCRIPTION
**Pull request by plokhotnyuk**

### CODE REVIEW

**Comment by PiotrFras** [comment]
Hi @plokhotnyuk,

thank you for your contribution!
Do you mind updating README, especially `Modules` and `Examples` sections?

Many thanks

**Comment by plokhotnyuk** [comment]
Hi, @PiotrFras,

Great!

I'll update both README sections in this PR.

File: jsoniter-scala/src/main/scala/com/ovoenergy/kafka/serialization/jsoniter_scala/package.scala [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package com.ovoenergy.kafka.serialization
+
+package object jsoniter_scala extends JsoniterScalaSerialization
```
Comment (PiotrFras) [inline jsoniter-scala/src/main/scala/com/ovoenergy/kafka/serialization/jsoniter_scala/package.scala]: I am not sure if I like the `_`. @filosganga What do you think? Is camel-case any better?

Comment (PiotrFras) [reply jsoniter-scala/src/main/scala/com/ovoenergy/kafka/serialization/jsoniter_scala/package.scala]: Or just drop `scala` suffix and make it `jsoniter`?

Comment (plokhotnyuk) [reply jsoniter-scala/src/main/scala/com/ovoenergy/kafka/serialization/jsoniter_scala/package.scala]: Jsoniter is a name of Java/Go project that inspired me to reuse their ideas for the Scala world:
http://jsoniter.com/

Comment (plokhotnyuk) [reply jsoniter-scala/src/main/scala/com/ovoenergy/kafka/serialization/jsoniter_scala/package.scala]: Snake case is recommended by official Java Tutorial: https://docs.oracle.com/javase/tutorial/java/package/namingpkgs.html

Using both upper and lower case chars in package names can lead for clashing on file systems that have case insensitive directories.

File: README.md [hunk 2]
```diff
@@ -83,7 +85,42 @@ val consumer = new KafkaConsumer(
 )
 ```
 
+## Jsoniter Scala example
```
Comment (filosganga) [inline README.md]: This file is generated, you should change the file in /doc/src/main/tut and then run
````bash
sbt tut
````

We should pointing out in the README file.

File: jsoniter-scala/src/main/scala/com/ovoenergy/kafka/serialization/jsoniter_scala/package.scala (hunk 1 above)
Comment (filosganga) [reply jsoniter-scala/src/main/scala/com/ovoenergy/kafka/serialization/jsoniter_scala/package.scala]: Why don't we keep the module name `jsoniter`, so we don't have to handle this situation?

**Comment by plokhotnyuk** [comment]
Jsoniter Scala does not depend on Jsoniter, it is like remix and mostly focused on compile-time generated codecs for binding, while original library support different data (AST/binding) & runtime (iteration/runtime-generated binding/reflection-based binding) models and their combinations.

I would prefer to leave that `_scala` suffix to simplify search sources of the library and its usage in the WWW.

**Pull request merged**